src/util/browser/event.js

```diff
@@ -42,8 +42,8 @@ var Event = {
   }
 };
 
-if (global.onunload !== undefined)
-  Event.on(global, 'unload', Event.detach, Event);
+if (global.onbeforeunload !== undefined)
```

Comment:
In [Coglan's response](https://github.com/faye/faye/issues/547#issuecomment-2704679359) he's concerned that using `beforeunload` would interfere with auto-disconncet behavior, which I think is [this code](https://github.com/zendesk/faye/blob/6b66a89b6b249f2fce68a87e399207d2a41e3adc/src/protocol/client.js#L60).

I think we should first try it his way, by just removing this to see if everything is working as intended.